pkg/daemon/server/service/isbs_query_service.go

```diff
@@ -94,6 +108,62 @@ func (is *isbSvcQueryService) GetBuffer(ctx context.Context, req *daemon.GetBuff
 	return resp, nil
 }
 
+// GetVertexInfo is used to query the metrics service and is used to obtain the processing rate of a given vertex for 1m, 5m and 15m.
+// In the future maybe latency will also be added here?
+// Should this method live here or maybe another file?
+func (is *isbSvcQueryService) GetVertexInfo(ctx context.Context, req *daemon.GetVertexRequest) (*daemon.GetVertexResponse, error) {
+	log := logging.FromContext(ctx)
+	resp := new(daemon.GetVertexResponse)
+
+	metricsPort := strconv.Itoa(v1alpha1.VertexMetricsPort)
+	pipelineVertex := fmt.Sprintf("%s-%s", is.pipeline.Name, *req.Vertex)
+	// We can query the metrics endpoint of the 0th pod to obtain this value.
+	// example: https://simple-pipeline-in-0.simple-pipeline-headless.svc.cluster.local:2469/metrics
+	url := fmt.Sprintf("https://%s-0.%s-headless.%s.svc.cluster.local:%s/metrics", pipelineVertex, pipelineVertex, *req.Namespace, metricsPort)
+
+	res, err := is.httpClient.Get(url)
+
+	if err != nil {
+		log.Errorf("Error reading the metrics endpoint: %s", err.Error())
+		return nil, err
+	}
+
+	// expfmt Parser from prometheus to parse the metrics
+	textParser := expfmt.TextParser{}
+	result, err := textParser.TextToMetricFamilies(res.Body)
+	if err != nil {
+		log.Errorf("Error in parsing to prometheus metric families: %s", err.Error())
+		return nil, err
+	}
+
+	processingRates := make([]*daemon.ProcessingRate, 0)
+	// Check if the resultant metrics list contains the processingRate, if it does look for the period label
+	if value, ok := result[v1alpha1.VertexProcessingRate]; ok {
+		metrics := value.GetMetric()
+		for _, metric := range metrics {
+			labels := metric.GetLabel()
+			for _, label := range labels {
```

Comment:
Let's assume the returned object is named `VertexAdditionalInfo`, Is it possible to return the obj with a property named `processingRates` in a `map[string] float` structure? The key is `1m`, `5m`, `15m` and `default`.

Reply:
Changed it to use map